components/registry-facade/pkg/registry/registry.go

```diff
@@ -299,9 +300,22 @@ func getRedisClient(cfg *config.RedisCacheConfig) (*redis.Client, error) {
 	defer cancel()
 
 	rdc := redis.NewClient(opts)
-	_, err := rdc.Ping(ctx).Result()
+
+	err := wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
+		Steps:    5,
+		Duration: 50 * time.Millisecond,
+		Factor:   2.0,
+		Jitter:   0.2,
+	}, func(ctx context.Context) (bool, error) {
+		_, err := rdc.Ping(ctx).Result()
+		if err != nil {
+			return false, xerrors.Errorf("cannot check Redis connection: %w", err)
```

Comment:
I'm not sure we want to return an error here within the backoff. The docs indicate it'll not retry if we return an error.

> It immediately returns an error if the condition returns an error, the context is cancelled or hits the deadline, or if the maximum attempts defined in backoff is exceeded (ErrWaitTimeout). If an error is returned by the condition the backoff stops immediately.

Maybe instead we should:
1. if err != nil, log the error as a warning, but do `return false, nil`. 
2. if err == nil, `return true, nil`
3. this way it'll keep retrying, and once it its the backoff step limit, it'll fail

A similar example:
https://github.com/gitpod-io/gitpod/blob/caa2889b76cb00dea812bfd9f8db099ccf25a9a2/components/registry-facade/pkg/registry/blob.go#L241-L245

Reply:
Fixed